unlock-app/src/__tests__/services/walletService.test.js

```diff
@@ -55,15 +58,64 @@ nock.emitter.on('no match', function(clientRequestObject, options, body) {
 })
 
 let walletService
+let config
 let providers
 
 describe('WalletService', () => {
+  let delay
   beforeEach(() => {
+    delay = require('../../utils/promises').delayPromise
```

Comment:
I still don't like with that ;)

Reply:
The mock for delayPromise is a `jest.fn()` so we need a new one for every test. That's all this does, is give us something to assert on. Is that kosher? :)

Reply:
This allows the tests to assert on the mock surrounding the promise function. Because we don't pass it as a parameter to the service, this is the only way to get a new one for every test.